### PR TITLE
chore: deprecate and add javadoc notes for methods to be removed in v7

### PR DESCRIPTION
**Pull request by dordsor21**

### CODE REVIEW

File: Core/src/main/java/com/plotsquared/core/components/ComponentPreset.java [hunk 1]
```diff
@@ -74,30 +74,65 @@ public static ComponentPreset deserialize(final @NonNull Map<String, Object> map
         );
     }
 
+    /**
+     * @deprecated {@link ComponentPreset} class will be replaced by a record in the next major update. This method will become
+     *         {@code ComponentPreset#component()}
+     */
+    @Deprecated(forRemoval = true, since = "TODO")
```
Comment (NotMyFault) [inline Core/src/main/java/com/plotsquared/core/components/ComponentPreset.java:81]: The same deprecation should be placed on the class itself too.

File: Core/src/main/java/com/plotsquared/core/location/ChunkWrapper.java [hunk 2]
```diff
@@ -23,8 +23,25 @@
 
 public class ChunkWrapper {
 
+    /**
+     * @deprecated {@link ChunkWrapper} class will be replaced by a record in the next major update. This variable will be
+     *         accessible using {@code ChunkWrapper#x()}
+     */
+    @Deprecated(forRemoval = true, since = "TODO")
```
Comment (NotMyFault) [inline Core/src/main/java/com/plotsquared/core/location/ChunkWrapper.java:30]: The same deprecation should be placed on the class itself too.

File: Core/src/main/java/com/plotsquared/core/plot/comment/PlotComment.java [hunk 3]
```diff
@@ -22,11 +22,45 @@
 
 public class PlotComment {
 
+    /**
+     * @deprecated {@link PlotComment} class will be replaced by a record in the next major update. This variable will be
+     *         accessible using {@code PlotComment#comment()}
+     */
+    @Deprecated(forRemoval = true, since = "TODO")
```
Comment (NotMyFault) [inline Core/src/main/java/com/plotsquared/core/plot/comment/PlotComment.java:29]: The same deprecation should be placed on the class itself too.

File: Core/src/main/java/com/plotsquared/core/util/FileBytes.java [hunk 4]
```diff
@@ -20,7 +20,18 @@
 
 public class FileBytes {
 
+    /**
+     * @deprecated {@link FileBytes} class will be replaced by a record in the next major update. This variable will be
+     *         accessible using {@code FileBytes#path()}
+     */
+    @Deprecated(forRemoval = true, since = "TODO")
```
Comment (NotMyFault) [inline Core/src/main/java/com/plotsquared/core/util/FileBytes.java:27]: The same deprecation should be placed on the class itself too.

File: Core/src/main/java/com/plotsquared/core/uuid/UUIDMapping.java [hunk 5]
```diff
@@ -36,10 +36,20 @@ public UUIDMapping(final @NonNull UUID uuid, final @NonNull String username) {
         this.username = username;
     }
 
+    /**
+     * @deprecated {@link UUIDMapping} class will be replaced by a record in the next major update. This method will become
+     *         {@code UUIDMapping#username()}
+     */
+    @Deprecated(forRemoval = true, since = "TODO")
```
Comment (NotMyFault) [inline Core/src/main/java/com/plotsquared/core/uuid/UUIDMapping.java:43]: The same deprecation should be placed on the class itself too.

**Review by SirYwell — COMMENTED** [review]
I'm not sure about the deprecations without existing replacement. Those will show up as warnings or errrors in tools, but developers can't do anything about it as there is no alternative (yet), and if they want to still provide support v6, they're stuck with those warnings.

For getters in records, I'd say we either
- just remove them in v7 without previous deprecation
- add fluent accessors as a replacement in v6 already
- add explicit getters to v7

What's a good solution might depend on the actual class, especially for classes that are meant to be used internally only, I'd just mark them as internal and just make them records in v7 without changing anything else. For classes that are meant to be consumed by API users, allowing to use fluent accessors in v6 already might be a better solution.

I don't have a strong opinion on the actual solution, but I don't think it#s useful for anyone to deprecate the methods that will only get replacements in v7.

**Comment by NotMyFault** [comment]
> add fluent accessors as a replacement in v6 already

That may be the solution with the least maintenance burden.

**Comment by NotMyFault** [comment]
I separated this PR into several ones, making it easier to build consensus: https://github.com/IntellectualSites/PlotSquared/pull/397, https://github.com/IntellectualSites/PlotSquared/pull/3977 and https://github.com/IntellectualSites/PlotSquared/pull/3976

**Pull request closed without merging**